### PR TITLE
Roll engine to f79fab1c97a2af91543984bcd656cb60b7ceb2db.

### PR DESCRIPTION
**Pull request by floitschG**
This patch is based on https://github.com/flutter/flutter/pull/18487.

### CODE REVIEW

**Review by mkustermann — APPROVED** [review]

**Comment by zanderso** [comment]
Did the auto-updater not try to update the matcher package, or did you back it out? I think @gspencergoog hit an issue with the deprecation of isInstanceOf from https://github.com/dart-lang/matcher/commit/6d8d0170256da2f332b7910263941016a4cf782a, and I saw the same issue when updating my roll CL. /cc @kevmoo

**Comment by kevmoo** [comment]
I know nothing of the `auto-updater`.

The deprecation is goodness, but I know flutter does not like deprecated usages...

**Comment by gspencergoog** [comment]
Yeah, you can't just update path to 1.6.1.  You have to run `flutter update-packages --force-upgrade` which will solve the transitive closure of all the versions for all the packages.  If you want, I can do this roll, since I already have the needed changes in a branch...

**Comment by gspencergoog** [comment]
Or, I can just send you a diff too.

File: packages/flutter_tools/test/base/build_test.dart [hunk 1]
```diff
@@ -406,7 +406,6 @@ void main() {
         '--url_mapping=dart:vmservice_io,${fs.path.join(skyEnginePath, 'sdk_ext', 'vmservice_io.dart')}',
         '--embedder_entry_points_manifest=$kVmEntrypoints',
         '--embedder_entry_points_manifest=$kIoEntries',
-        '--dependencies=${fs.path.join(outputPath, 'snapshot.d')}',
```
Comment (gspencergoog) [inline packages/flutter_tools/test/base/build_test.dart]: What's the reason for this change?  Is it because that's the default?

Comment (zanderso) [reply packages/flutter_tools/test/base/build_test.dart]: gen_snapshot now fails when asked to generate a depfile in Dart 2 mode since it can't do it.

**Comment by gspencergoog** [comment]
I talked with Florian: I'm going to just send a new PR with my changes and these in it.

**Comment by gspencergoog** [comment]
Closing this in favor of this [newer roll](https://github.com/flutter/flutter/pull/18575).
